straxen/url_config.py

```diff
@@ -329,3 +334,66 @@ def open_neural_net(model_path: str, **kwargs):
         tar = tarfile.open(model_path, mode="r:gz")
         tar.extractall(path=tmpdirname)
         return tf.keras.models.load_model(tmpdirname)
+
+
+@URLConfig.register('itp_dict')
+def get_itp_dict(loaded_json,
+                 run_id=None,
+                 time_key='time',
+                 itp_keys='correction',
+                 **kwargs) -> typing.Union[np.ndarray, typing.Dict[str, np.ndarray]]:
+    """
+    Interpolate the dictionary at the start time that is queried from
+    a run-id.
+
+    The loaded JSON (dictionary), should have at least one collum `time`
+    and has other columns given by `itp_dict_keys`.
+
+    :param loaded_json: a dictionary with a time-series
+    :param run_id: run_id
+    :param time_key: key that gives the timestamps
+    :param itp_keys: which keys from the dict to read. Should be
+        comma (',') separated!
+
+    :return: Interpolated values of dict at the start time, either
+        returned as an np.ndarray (single value) or as a dict
+        (multiple itp_dict_keys)
+    """
+    keys = strax.to_str_tuple(itp_keys.split(','))
+    times = loaded_json[time_key]
+    if not all(k in loaded_json for k in keys):
+        raise ValueError(f'One or more of {itp_keys} are not in {loaded_json.keys()}')
```

Comment:
The check should go before asking for any item and the time-key should be checked too.

Reply:
Done